client-sdk/go/types/transaction.go

```diff
@@ -2,17 +2,26 @@
 package types
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
+	"io"
+	"reflect"
+	"sync"
 
 	"github.com/oasisprotocol/oasis-core/go/common/cbor"
 	"github.com/oasisprotocol/oasis-core/go/common/crypto/hash"
+	"github.com/oasisprotocol/oasis-core/go/common/prettyprint"
 	"github.com/oasisprotocol/oasis-core/go/common/quantity"
-
 	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature"
 )
 
 // SignatureContextBase is the transaction signature domain separation context base.
-var SignatureContextBase = []byte("oasis-runtime-sdk/tx: v0")
+var (
+	SignatureContextBase = []byte("oasis-runtime-sdk/tx: v0")
+
+	registeredMethods sync.Map
```

Comment:
can you add a public accessor, this would be nice for generating APIs

Reply:
Good idea.